psh/test_ls.py

```diff
@@ -39,7 +39,27 @@ def prepare_testdir(p, files):
     psh.assert_cmd(p, f'mkdir {TEST_DIR_BASIC}/dir', '', msg)
 
 
-def create_pattern(files, separate_lines=False):
+def create_unordered_pattern(files):
+    """ Create regex pattern for ls output
+    containing all files from the `files` list in any order """
+    pattern = r''
+    pattern += '(' + OPTIONAL_CONTROL_CODE + '('
+    for file in files:
+        pattern += file
+        if file == files[-1]:
+            pattern += ')'
+        else:
+            pattern += '|'
```

Comment:
One can use .join function

https://www.w3schools.com/python/ref_string_join.asp

Reply:
the same applies for ordered function

Reply:
Thanks, applied.